cuda/matrix/csr_kernels.cu

```diff
@@ -824,6 +917,95 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_CSR_CONJ_TRANSPOSE_KERNEL);
 
 
+namespace kernel {
+
+
+template <typename IndexType>
+__global__
+    __launch_bounds__(default_block_size) void calculate_max_nnz_per_block(
+        size_type num_rows, const IndexType *__restrict__ row_ptrs,
+        IndexType *__restrict__ result)
+{
+    const auto tidx = threadIdx.x + blockIdx.x * blockDim.x;
+    const auto num_threads = blockDim.x * gridDim.x;
+
+    __shared__ IndexType block_result[default_block_size];
+
+    block_result[threadIdx.x] =
+        (tidx < num_rows) ? row_ptrs[tidx + 1] - row_ptrs[tidx] : 0;
+
+    for (auto i = num_threads; i < num_rows; i += num_threads) {
+        block_result[threadIdx.x] =
+            max(block_result[threadIdx.x], row_ptrs[i + 1] - row_ptrs[i]);
```

Comment:
It seems that you take block_result to compare the same things in different threads.
I guess you mean `for (auto i = num_threads + tidx; ....)` here

Reply:
I think this part of the code is now gone (replaced by a `reduce` call if I remember correctly), but I am not 100% sure.
@fritzgoebel do you remember where this part was actually from?

Reply:
Yes, this part of the code is gone.